### ISSUE TITLE
Compiler allows marker trait bounds to be omitted in implementations when using RPITIT

### ISSUE THREAD

**Issue by jrose-signal** [rust-lang/rust#151433]
I tried this code:

```rust
pub trait Example {
    fn test() -> impl Eq + Ord + Send;
}

impl Example for u8 {
    fn test() -> impl Eq + Ord {
        "example"
    }
}
```

I expected an error or at least a warning, but the code compiled with no diagnostics. Omitting `Eq` or `Ord` from the implementing function does produce an error.

### Meta
Tested with both stable (1.92) and nightly (2026-01-19) in the Playground.

**Comment by workingjubilee** [comment]
Yep. Builds (incorrectly):

```rust
use std::hash::Hash;

pub trait Example {
    fn test() -> impl Ord + Hash + Send;
}

impl Example for u8 {
    fn test() -> impl Ord + Hash {
        "example"
    }
}
```

Errors (correctly):
```rust
use std::hash::Hash;

pub trait Example {
    fn test() -> impl Ord + Hash + Send;
}

impl Example for u8 {
    fn test() -> impl Ord + Send {
    //           ^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `impl Ord + Send`
    // error[E0277]: the trait bound `impl Ord + Send: Hash` is not satisfied
        "example"
    }
}

impl Example for u16 {
    fn test() -> impl Hash + Send {
    //                ^^^^^^^^^^^ the trait `Ord` is not implemented for `impl Hash + Send`
    // error[E0277]: the trait bound `impl Hash + Send: Ord` is not satisfied
        "example"
    }
}
```

You're good, just being annoyingly precise about the pass/fail conditions in case this is one of those issues that hangs around for a while.

**Comment by QuineDot** [comment]
I can understand how this came about, since auto-traits intentionally leak through RPIT[IT]s but other bounds do not.  (The opaque does unconditionally implement `Send`, and callers can rely on that, even though it's invisible.  They cannot rely on `Hash` if you omit it.)  But I also agree it'd be an improvement to require repeating the (required) commitment.

The error note when the auto-trait is conditional is also suboptimal, but amusing.
```rust
pub trait Example {
    fn test<T: Default + Eq + Ord>() -> impl Eq + Ord + Send;
}

impl Example for u8 {
    fn test<T: Default + Eq + Ord>() -> impl Eq + Ord {
        T::default()
    }
}
```
```rust
error[E0277]: `T` cannot be sent between threads safely
 --> src/lib.rs:6:41
  |
6 |     fn test<T: Default + Eq + Ord>() -> impl Eq + Ord {
  |                                         ^^^^^^^^^^^^^
  |                                         |
  |                                         `T` cannot be sent between threads safely
  |                                         within this `impl Eq + Ord`
  |
note: required because it appears within the type `impl Eq + Ord`
 --> src/lib.rs:6:41
  |
6 |     fn test<T: Default + Eq + Ord>() -> impl Eq + Ord {
  |                                         ^^^^^^^^^^^^^
note: required by a bound in `Example::test::{anon_assoc#0}`
```
